super-csv/src/main/java/org/supercsv/io/Tokenizer.java

```diff
@@ -228,31 +231,65 @@ else if( c == quoteChar ) {
 					potentialSpaces = 0;
 					currentColumn.append(c);
 				}
-				
+
 			} else {
-				
+
 				/*
 				 * QUOTE_MODE (within quotes).
 				 */
-				
-				if( c == quoteChar ) {
+
+				if( c == quoteEscapeChar ) {
 					int nextCharIndex = charIndex + 1;
 					boolean availableCharacters = nextCharIndex < line.length();
 					boolean nextCharIsQuote = availableCharacters && line.charAt(nextCharIndex) == quoteChar;
+					boolean nextCharIsEscapeQuoteChar = availableCharacters && line.charAt(nextCharIndex) == quoteEscapeChar;
+
 					if( nextCharIsQuote ) {
 						/*
-						 * An escaped quote (""). Add a single quote, then move the cursor so the next iteration of the
-						 * loop will read the character following the escaped quote.
+						 * An escaped quote (e.g. "" or \"). Skip over the escape char, and add
+						 * the following quote char as part of the column;
+						 */
+						charIndex++;
+						currentColumn.append(line.charAt(nextCharIndex));
+					} else if( nextCharIsEscapeQuoteChar ) {
+						/*
+						 * A double escape (normally \\). Save the escape char, then continue to
+						 * next character.
 						 */
 						currentColumn.append(c);
 						charIndex++;
-						
-					} else {
+					} else if( quoteEscapeChar == quoteChar ) {
```

Comment:
This is the situation for a field that looks like this: `"col1","col2"` with both `quoteChar` and `quoteEscapeChar` as `"`.  For the second quote in that string, the one ending col1, if we don't have this branch then the quote is treated as an escape char that isn't before another escape char or quote char, and then added as part of the value in the first column.

We have to have this branch in order to handle the specific case where `quoteChar == quoteEscapeChar`